onnx/defs/schema.cc

```diff
@@ -96,9 +96,10 @@ void OpSchema::CheckInputOutputType(struct InferenceContext& ctx) const {
     }
     if (param.GetIsHomogeneous()) {
       const auto& type_proto = Utils::DataTypeUtils::ToType(*param_type);
-      if (type_constraints.find(type_str) == type_constraints.end()) {
+      auto it = type_constraints.find(type_str);
```

Comment:
Assuming we will not find many inconsistent duplicate registrations you can also combine insert/find and eliminate double lookup if you want to be real efficient. Something like:
`auto p = type_constraints.emplace(type_str, *type_proto);
if(!p.second) {
// failed to insert due to a duplicate, check consistency
if(p.first.second != *type_proto) fail()
}
`

Reply:
nice trick!